src/RoslynInsertionTool/RoslynInsertionTool/RoslynInsertionTool.Git.cs

```diff
@@ -238,15 +239,67 @@ private static void StageFiles(BuildVersion newRoslynVersion,CancellationToken c
             var repositoryStatus = Enlistment.RetrieveStatus();
             if (repositoryStatus.IsDirty)
             {
+                cancellationToken.ThrowIfCancellationRequested();
+
                 var filesToStage = repositoryStatus
                     .Where(item => item.State != FileStatus.Unaltered && item.State != FileStatus.Ignored)
-                    .Select(item => item.FilePath);
+                    .Select(item => item.FilePath).ToList();
 
                 cancellationToken.ThrowIfCancellationRequested();
-                Log.Info($"Staging {filesToStage.Count()} file(s)");
-                var watch = Stopwatch.StartNew();
-                Enlistment.Stage(filesToStage, GetStageOptions());
-                Log.Trace($"Staging took {watch.Elapsed.TotalSeconds} seconds");
+                if (!isWhitespace(Enlistment.Diff.Compare<Patch>(filesToStage)))
+                {
+                    Log.Info($"Staging {filesToStage.Count()} file(s)");
+                    var watch = Stopwatch.StartNew();
+                    Enlistment.Stage(filesToStage, GetStageOptions());
+                    Log.Trace($"Staging took {watch.Elapsed.TotalSeconds} seconds");
+                }
+                else
+                {
+                    Log.Info("Only whitespace changes found");
+                }
+            }
+
+            bool isWhitespace(Patch p)
+            {
+                var adds = new StringBuilder();
+                var removes = new StringBuilder();
+                foreach (var change in p)
+                {
+                    if (change.Status != ChangeKind.Modified)
+                    {
+                        return false;
+                    }
+
+                    using (var reader = new StringReader(change.Patch))
+                    {
+                        string line;
+                        while ((line = reader.ReadLine()) != null)
+                        {
+                            if (line.StartsWith("---") || line.StartsWith("+++"))
+                            {
+                                continue;
+                            }
+
+                            if (line.StartsWith("+"))
+                            {
+                                adds.Append(line.TrimStart('+').Trim());
```

Comment:
.TrimStart won't trim just the first + but rather all of them, so if you had a line going from + to - (as actual text content), this will mis-fire. :smile:

Reply:
Good point. I will just trim the first character.